src/client.rs

```diff
@@ -174,15 +234,49 @@ pub mod future {
         }
     }
 
-    impl<Req, Resp, E> FnOnce<(TcpStream,)> for MultiplexConnect<Req, Resp, E>
+    impl<Req, Resp, E, I> FnOnce<(I,)> for MultiplexConnect<Req, Resp, E>
         where Req: Serialize + Sync + Send + 'static,
               Resp: Deserialize + Sync + Send + 'static,
-              E: Deserialize + Sync + Send + 'static
+              E: Deserialize + Sync + Send + 'static,
+              I: Into<StreamType>
     {
         type Output = Client<Req, Resp, E>;
 
-        extern "rust-call" fn call_once(self, (tcp,): (TcpStream,)) -> Client<Req, Resp, E> {
-            Client::new(Proto::new().bind_client(&self.0, tcp))
+        extern "rust-call" fn call_once(self, (stream,): (I,)) -> Self::Output {
+            Client::new(Proto::new().bind_client(&self.0, stream.into()))
+        }
+    }
+
+    /// Provides the connection Fn impl for Tls
+    #[doc(hidden)]
+    pub struct ConnectFn {
```

Comment:
Oh yes, awesome!

Rust was originally complaining about private type in a public interface, but that was with an older version of the `Connect` trait. Making this private seems okay now. I like that it can be private again as well.